### PR TITLE
[LLAMA] update pattern to consider initializers

### PR DESCRIPTION
**Pull request by dsikka**
For the quantized models, the Slice nodes that need to be updated have input values stored as initializers, while for the dense models, they're stored as constants. This PR updates to check for both.

### CODE REVIEW

File: src/sparseml/exporters/transforms/kv_cache/transforms_llama.py [hunk 1]
```diff
@@ -103,13 +103,24 @@ def update_slice_nodes_for_positions_input(self, model: ModelProto) -> ModelProt
 
         nodes_found = 0
         for node in model.graph.node:
+            valid_node = False
+
             if node.op_type == "Slice":
+                init = get_init_by_name(model, node.input[0])
```
Comment (dsikka) [inline src/sparseml/exporters/transforms/kv_cache/transforms_llama.py:109]: update if

File: src/sparseml/exporters/transforms/kv_cache/transforms_llama.py [hunk 2]
```diff
@@ -103,13 +103,24 @@ def update_slice_nodes_for_positions_input(self, model: ModelProto) -> ModelProt
...
+                    valid_node = True
+                elif data_parent is not None and len(data_parent.input) == 0:
+                    valid_node = True
+
+                if valid_node:
                     nodes_found += 1
                     node.input[2] = self.SLICE_MAX_INT_NAME
                     self.log_match(node)
 
+        assert nodes_found == 64
```
Comment (dsikka) [inline src/sparseml/exporters/transforms/kv_cache/transforms_llama.py:123]: check and warning

**Review by dsikka — COMMENTED** [review]
address comments

File: src/sparseml/exporters/transforms/kv_cache/transforms_llama.py [hunk 3]
```diff
@@ -103,14 +104,31 @@ def update_slice_nodes_for_positions_input(self, model: ModelProto) -> ModelProt
...
+            if valid_node:
+                nodes_found += 1
+                node.input[2] = self.SLICE_MAX_INT_NAME
+                self.log_match(node)
+
+        valid_node_counts = [64, 80]
+        if nodes_found not in valid_node_counts:
+            warnings.warn(
+                f"Number of Slice nodes updated {nodes_found} does not match the "
+                f"expected values {valid_node_counts} for the 7 billion and 13 billion "
```
Comment (bfineran) [inline src/sparseml/exporters/transforms/kv_cache/transforms_llama.py:127]:
```suggestion
                f"expected values {valid_node_counts} for the 7 billion or 13 billion "
```

**Review by bfineran — APPROVED** [review]

**Review by bfineran — APPROVED** [review]

**Pull request merged**